squeal-postgresql/src/Squeal/PostgreSQL/Binary.hs

```diff
@@ -831,6 +845,20 @@ newtype Only x = Only { fromOnly :: x }
 instance Generic (Only x)
 instance HasDatatypeInfo (Only x)
 
+-- | A refined text type for use with 'varchar' and 'char'.
+-- The constructor is not exposed. You have to use @constructVarChar@ and @unVarChar@.
+newtype VarChar (n :: Nat) = VarChar Strict.Text
```

Comment:
Can you move this all to the `PG` module next to other storage newtypes and add a `type instance PG (VarChar n) = 'PGvarchar n`?